New syntax for `dotnet new`

vlada-shubina opened the pull request:
## Summary
- changed `dotnet new` articles headers
- added information about new `dotnet new` syntax available from .NET SDK 7.0.100 Preview 2

vlada-shubina committed:
changed dotnet new articles headers

vlada-shubina committed:
added new syntax statement

vlada-shubina committed:
split examples to related articles

vlada-shubina commented:
@DamianEdwards do you want to update https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new-sdk-templates to include `--use-program-main` flag? (as in https://github.com/dotnet/docs/pull/28857/commits/3c742fcda94c179e6e09a0ddfb3e8e50cdfbf38c)

baronfel commented on includes/dotnet-new-7-0-syntax.md:
```diff
@@ -0,0 +1,20 @@
+- `--update-apply` option became `update` subcommand.
+- to use `--update-check`, use `update` subcommand with `--check` option.
+
+Other options available before are still available to use with respective subcommands.
```
```suggestion
Other options available before are still available to use with their respective subcommands.
```

tdykstra replied:
```suggestion
Other options that were available before are still available to use with their respective subcommands.
```

baronfel reviewed (approved):
This looks great :) A couple small grammar suggestions are all I had.

vlada-shubina commented:
@tdykstra i'm not sure about current `include`, would it be better to include it as a note, so syntax differs from main article content? Is there a similar syntax for docs?

baronfel commented on docs/core/tools/dotnet-new-list.md:
````diff
@@ -22,6 +22,23 @@ dotnet new [<TEMPLATE_NAME>] -l|--list [--author <AUTHOR>] [-lang|--language {"C
+- Show help for list subcommand
+
+```
+dotnet new list --help
````
can you update [this example](https://github.com/dotnet/docs/blob/9afd7ac45aa807749e4bc07938c3c2b1e13f0c35/docs/core/tools/dotnet-new-sdk-templates.md?plain=1#L13) on the sdk templates page as well?

vlada-shubina replied:
We decided not to update the other articles yet. This command will still work.
I haven't added a note to this article however, do you think we should add it here as well?

tdykstra replied:
I think it would be good to add it here as well.

tdykstra commented on includes/dotnet-new-7-0-syntax.md:
```diff
@@ -0,0 +1,20 @@
+
+Starting .NET SDK 7.0.100  Preview 2 `dotnet new` syntax has changed:
+
+- `--list`, `--search`, `--install` `--uninstall` options became `list`, `search`, `install` and `uninstall` subcommand.
```
```suggestion
- The `--list`, `--search`, `--install`, and `--uninstall` options became `list`, `search`, `install` and `uninstall` subcommands.
```

tdykstra commented on includes/dotnet-new-7-0-syntax.md:
```diff
@@ -0,0 +1,20 @@
+- to use `--update-check`, use `update` subcommand with `--check` option.
+
+Other options available before are still available to use with respective subcommands.
+There is a separate help for each subcommand avilable via `-h` or `--help` option: `dotnet new <subcommand> --help`, that lists all supported options for the subcommand.
```
```suggestion
Separate help for each subcommand is available via the `-h` or `--help` option: `dotnet new <subcommand> --help` lists all supported options for the subcommand.
```

tdykstra commented on includes/dotnet-new-7-0-syntax.md:
```diff
@@ -0,0 +1,20 @@
+Other options available before are still available to use with respective subcommands.
+There is a separate help for each subcommand avilable via `-h` or `--help` option: `dotnet new <subcommand> --help`, that lists all supported options for the subcommand.
+
+Additionally, the tab completion is now available for `dotnet new`.
```
```suggestion
Additionally, tab completion is now available for `dotnet new`.
```

tdykstra commented on includes/dotnet-new-7-0-syntax.md:
```diff
@@ -0,0 +1,20 @@
+There is a separate help for each subcommand avilable via `-h` or `--help` option: `dotnet new <subcommand> --help`, that lists all supported options for the subcommand.
+
+Additionally, the tab completion is now available for `dotnet new`.
+To activate tab completion for .NET SDK, follow [the guideline](../docs/core/tools/enable-tab-autocomplete.md).
```
```suggestion
To activate tab completion for the .NET SDK, see [Enable tab completion](../docs/core/tools/enable-tab-autocomplete.md).
```

tdykstra commented on docs/core/tools/dotnet-new-install.md:
```diff
@@ -31,6 +31,23 @@ For example, if you install the template package using `dotnet new --install` in
+
+Examples:
+
+- Show help for install subcommand
```
```suggestion
- Show help for `install` subcommand
```

tdykstra commented on docs/core/tools/dotnet-new-list.md:
```diff
@@ -22,6 +22,23 @@ dotnet new [<TEMPLATE_NAME>] -l|--list [--author <AUTHOR>] [-lang|--language {"C
+
+Examples:
+
+- Show help for list subcommand
```
```suggestion
- Show help for `list` subcommand
```

vlada-shubina committed:
Apply suggestions from code review

Co-authored-by: Chet Husk <baronfel@users.noreply.github.com>

tdykstra commented on docs/core/tools/dotnet-new-search.md:
```diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
```
```suggestion
Examples of the new syntax:
```

tdykstra commented on docs/core/tools/dotnet-new-search.md:
```diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
+
+Examples:
+
+- Show help for search subcommand
```
```suggestion
- Show help for the `search` subcommand.
```

tdykstra commented on docs/core/tools/dotnet-new-search.md:
````diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
+dotnet new search --help
+```
+
+- Search for all templates available on NuGet.org matching the we substring and supporting the F# language
````
```suggestion
- Search for all templates available on NuGet.org matching the "we" substring and supporting the F# language
```

tdykstra commented on docs/core/tools/dotnet-new-uninstall.md:
```diff
@@ -21,6 +21,30 @@ dotnet new --uninstall <PATH|NUGET_ID>
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
```
```suggestion
Examples of the new syntax:
```

tdykstra commented on docs/core/tools/dotnet-new-search.md:
````diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
+dotnet new search we --language "F#"
+```
+
+
````
```suggestion
```

tdykstra commented on docs/core/tools/dotnet-new-uninstall.md:
```diff
@@ -21,6 +21,30 @@ dotnet new --uninstall <PATH|NUGET_ID>
+
+Examples:
+
+- Show help for uninstall subcommand
```
```suggestion
- Show help for the `uninstall` subcommand.
```

tdykstra commented on docs/core/tools/dotnet-new-uninstall.md:
````diff
@@ -21,6 +21,30 @@ dotnet new --uninstall <PATH|NUGET_ID>
+dotnet new uninstall Microsoft.Azure.WebJobs.ProjectTemplates
+```
+
+
````
```suggestion
```

tdykstra commented on docs/core/tools/dotnet-new-update.md:
```diff
@@ -26,6 +26,31 @@ dotnet new --update-apply
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
```
```suggestion
Examples of the new syntax:
```

tdykstra commented on docs/core/tools/dotnet-new-update.md:
```diff
@@ -26,6 +26,31 @@ dotnet new --update-apply
+
+Examples:
+
+- Show help for update subcommand
```
```suggestion
- Show help for the `update` subcommand.
```

tdykstra commented on docs/core/tools/dotnet-new-install.md:
````diff
@@ -31,6 +31,23 @@ For example, if you install the template package using `dotnet new --install` in
+- Show help for install subcommand
+
+```
+dotnet new install --help
````
everywhere there's a block like this introduced by a bullet point, the block should be indented two spaces so the rendered example text will appear at the same indentation as the bullet text.

vlada-shubina replied:
fixed

tdykstra reviewed (approved):
Looks great, I just made some grammar and formatting suggestions.

tdykstra commented:
> @tdykstra i'm not sure about current `include`, would it be better to include it as a note, so syntax differs from main article content? Is there a similar syntax for docs?

That's a good idea, but then the new-syntax examples should be in the same format.  Syntax is here: https://review.docs.microsoft.com/en-us/help/contribute/markdown-reference?branch=main#alerts-note-tip-important-caution-warning

vlada-shubina committed:
Apply suggestions from code review

Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>

vlada-shubina committed:
indentation for examples

vlada-shubina committed:
fix that was not applied automatically

vlada-shubina committed:
formatted include text as the note

vlada-shubina commented:
> added information about new console option --use-program-main available from .NET SDK 6.0.300

This edit was moved to https://github.com/dotnet/docs/pull/28858 together with web templates changes.

tdykstra commented:
> @tdykstra @baronfel I changed the paragraph to be a note, and also added it to `dotnet new <TEMPLATE>` article. Please check it, I prefer note style.

It makes sense to render it as a note. But it looks like the examples indentation was fixed only for `dotnet new list`.

vlada-shubina commented:
> > @tdykstra @baronfel I changed the paragraph to be a note, and also added it to `dotnet new <TEMPLATE>` article. Please check it, I prefer note style.
> 
> It makes sense to render it as a note. But it looks like the examples indentation was fixed only for `dotnet new list`.

It looks i missed an extra space in note, thanks for catching - will fix tomorrow.

vlada-shubina commented:
> > > @tdykstra I changed the paragraph to be a note, and also added it to `dotnet new <TEMPLATE>` article. Please check it, I prefer note style.
> > 
> > 
> > It makes sense to render it as a note. But it looks like the examples indentation was fixed only for `dotnet new list`.
> 
> It looks i missed an extra space in note, thanks for catching - will fix tomorrow.

Done - all pages have correct indentation now.
